src/headless/plugins/blocking/api.js

```diff
@@ -4,97 +4,103 @@ import { _converse, api, converse } from '@converse/headless/core.js';
 const { Strophe, $iq, sizzle, u } = converse.env;
 
 export default {
+
+    /**
+     * Checks if XEP-0191 is supported
+     */
+    async isBlockingAvailable () {
+        const has_feature = await api.disco.supports(Strophe.NS.BLOCKING, _converse.domain);
+        if (!has_feature) {
+            log.info("XEP-0191 not supported, no blocklist available");
+            return false;
+        }
+        log.debug("XEP-0191 available");
+        return true;
+    },
     /**
      * Retrieves the blocklist held by the logged in user at a JID by sending an IQ stanza.
-     * Saves the model variable _converse.blocked.set
      * @private
      * @method api.refreshBlocklist
      */
     async refreshBlocklist () {
```

Comment:
The API function names don't conform to the convention used in Converse.

The API is broken up into namespaces with short function names.

Look for example at the `disco` API: https://github.com/conversejs/converse.js/blob/master/src/headless/plugins/disco/api.js

There is a `disco` namespace, and then functions inside it.

The same needs to be done here. A `blocklist` namespace needs to be added, and then the functions here should be put inside that namespace and renamed.

So `api.isBlockingAvailable()` becomes `api.blocklist.available()`.
`api.blockUser()` becomes `api.blocklist.add()` and `api.unblockUser()` becomes `api.blocklist.remove()`.

And so forth.

Reply:
What about `_converse.blocked`? Should we put that in a namespace too, even if it's not a method?
Like `_converse.disco._identities`, should be `_converse.blocking.blocked`

Reply:
:+1:  Yeah, it's better to namespace it to avoid potential name clashes with other (e.g. 3rd party) plugins.